### ISSUE TITLE
vSphere provider wishlist

### ISSUE THREAD

**Issue by saptaji** [terraform-providers/terraform-provider-vsphere#335]
I have this wishlist for vSphere provider.

1. Snapshot Data Sources
    Want to be able to retrieve a list of snapshots for a VM so we can switch around which snapshot need to be active at a time.
    Use case: we have 'build' and 'test' servers and a couple of snapshots on those servers. 
    We activate a certain snapshot depend on what application that needs to be built or test.

2. CDROM
    Need to be able to create a CDROM without attaching to an iso.
    At the current state, we have to specify the 'datastore_id' and 'path.
    For Windows, we need to have a CDROM device to install VMWare tools.
    Don't want to attach an iso when creating CDROM device; just need an empty CDROM and let vSphere manage the VMWare tools installation/upgrade.

    Better yet: CDROM device will be created even without specifying it as part of the basic VM creation.

3. Generate New Security ID (SID) option
    For Windows, we need to have 'Generate New Security ID (SID)' option so we can use it in the ADDS (production) environment.
    Basically, we "can not" have servers with the same SID in the ADDS environment.

FYI, OS customization for Windows Server 2008 didn't work but don't bother since with PowerCLI also has the same problem.

**Comment by vancluever** [comment]
Hey @saptaji, just wanted to address these point-by-point:

## VM Snapshot Management/Data Sources

Unfortunately, I don't think this is a good fit for the vSphere provider and it will probably never get implemented. We already rejected a `vsphere_virtual_machine_snapshot_revert` resource in #107 that would ultimately be the end result of such a data source (you check there for some more detail on exactly why I don't think snapshots have a place in the TF vSphere provider).

Instead - for what it seems like you are trying to achieve - the recommended and supported pattern by the HashiCorp group of tools (inpartiuclarly Packer and Terraform) is to build separate templates for each individual purpose and deploy off of those. You can also use [provisioners](https://www.terraform.io/docs/provisioners/index.html) and configuration management tools to bring a more generic base image up to a specific configuration post-deployment.

## CDROM Without an ISO

This has already been entered as a feature request and you can track it in #300.

## Windows SID Re-generation on Customization

If this is not happening already, it's a bug. Rather than track it in something that's more of a question/meta-issue, I've opened up #336 to track this. Note that if you are not customizing, there might not be much we can do here (and the end result of #336 might be that) as we can't tell a VM to customize itself it the user has not requested it. You may have to manually sysprep and generalize the machine yourself.

-----

Hope this helps! Again, you can track the actionable items on the last two items in their respective issues.
